dask/array/ghost.py

```diff
@@ -351,30 +346,98 @@ def ghost(x, depth, boundary):
                              "with a larger chunk size or a chunk size that\n"
                              "more evenly divides the shape of your array." %
                              (d, min(c)))
-    x2 = boundaries(x, depth, boundary)
-    x3 = ghost_internal(x2, depth)
-    trim = dict((k, v*2 if boundary.get(k, None) is not None else 0)
-                for k, v in depth.items())
+    x2 = boundaries(x, depth2, boundary2)
+    x3 = ghost_internal(x2, depth2)
+    trim = dict((k, v*2 if boundary2.get(k, 'none') != 'none' else 0)
+                for k, v in depth2.items())
     x4 = chunk.trim(x3, trim)
     return x4
 
 
 def map_overlap(x, func, depth, boundary=None, trim=True, **kwargs):
+    """
+    Map a function over a dask array with ghost cells.
+
+    Parameters
+    ----------
+
+    x: da.Array
+        A dask array
+    func: function
+        The function applied to each block. It must accept a single numpy array.
+    depth: int, tuple, dict
+        The depth of the ghosting on each axis.
+    boundary: int, tuple, dict
+        The boundary condition on each axis. Options are 'reflect', 'periodic',
+        'nearest', 'none', an integer will fill the boundary with that integer.
+
+    >>> import numpy as np
+    >>> import dask.array as da
+    >>> x = np.arange(16).reshape((4, 4))
+    >>> d = da.from_array(x, chunks=(2, 2))
+    >>> d.map_overlap(lambda x: x + x.size, depth=1).compute()
+    array([[16, 17, 18, 19],
+           [20, 21, 22, 23],
+           [24, 25, 26, 27],
+           [28, 29, 30, 31]])
+
+    >>> func = lambda x: x + x.size
+    >>> depth = {0: 1, 1: 1}
+    >>> boundary = {0: 'reflect', 1: 'none'}
+    >>> map_overlap(d, func, depth, boundary).compute()  # doctest: +NORMALIZE_WHITESPACE
+    array([[ 12.,  13.,  14.,  15.],
+           [ 16.,  17.,  18.,  19.],
+           [ 20.,  21.,  22.,  23.],
+           [ 24.,  25.,  26.,  27.]])
+    """
```

Comment:
Note that there is maybe a second docstring at `Array.map_overlap`.  We might want to mutate that method to have this docstring or otherwise combine them.